app/src/main/java/org/simple/clinic/drugs/selection/custom/CustomDrugEntryUpdate.kt

```diff
@@ -67,6 +67,7 @@ class CustomDrugEntryUpdate : Update<CustomDrugEntryModel, CustomDrugEntryEvent,
       prescription: PrescribedDrug
   ): Next<CustomDrugEntryModel, CustomDrugEntryEffect> {
     val frequency = DrugFrequency.fromMedicineFrequency(prescription.frequency)
+    val cursorPosition = if (prescription.dosage != null) cursorPositionFromDosage(prescription.dosage) else 0
```

Comment:
I see why we have 2 conditions now. Apart from having `cursorPositionFromDosage` usage twice? Is there some other reason that I'm missing? 
If not, can we have one condition for these into their individual functions.